schema/pod.go

```diff
@@ -116,6 +116,11 @@ func (al AppList) assertValid() error {
 			return fmt.Errorf(`duplicate apps of name %q`, a.Name)
 		}
 		seen[a.Name] = true
+		if a.App != nil {
```

Comment:
I think I'm dumb, I'm really not seeing what would cause this panic (why is .App being dereferenced? the go marshaller should skip nil pointers). Can you explain it to me?